File: backend/app/models/event.py
import uuid

from sqlalchemy import String, DateTime
from sqlalchemy.orm import Mapped, mapped_column

from app.database import Base


class Event(Base):
    __tablename__ = "events"

    id: Mapped[str] = mapped_column(String, primary_key=True, default=lambda: str(uuid.uuid4()))
    title: Mapped[str] = mapped_column(String, nullable=False)
    start: Mapped[str] = mapped_column(DateTime, nullable=False)
    end: Mapped[str] = mapped_column(DateTime, nullable=False)
    source: Mapped[str] = mapped_column(String, nullable=False)  # work1 | work2 | personal | generated
    location: Mapped[str | None] = mapped_column(String, nullable=True)
    notes: Mapped[str | None] = mapped_column(String, nullable=True)
    external_id: Mapped[str | None] = mapped_column(String, nullable=True)

File: frontend/src/types/event.ts
// Calendar event types

export type CalendarSource = 'work1' | 'work2' | 'personal' | 'generated'

export interface CalendarEvent {
  id: string
  title: string
  start: string // ISO datetime
  end: string
  source: CalendarSource
  location?: string
  notes?: string
}

export interface RecurringEvent extends CalendarEvent {
  recurrenceRule: string // RRULE string
}
